cmd/gofakes3/main.go

```diff
@@ -2,30 +2,81 @@ package main
 
 import (
 	"flag"
+	"fmt"
 	"log"
 	"net"
 	"net/http"
 
 	"github.com/johannesboyne/gofakes3"
+	"github.com/johannesboyne/gofakes3/backend/s3bolt"
+	"github.com/johannesboyne/gofakes3/backend/s3mem"
 )
 
 func main() {
-	db := flag.String("db", "locals3.db", "Database path / name")
-	port := flag.String("port", ":9000", "Port to run the service")
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	var (
+		db      string
+		host    string
+		backend string
+		bucket  string
+	)
+
+	flag.StringVar(&db, "db", "locals3.db", "Database path / name when using bolt backend")
+	flag.StringVar(&host, "host", ":9000", "Host to run the service")
+	flag.StringVar(&backend, "backend", "", "Backend to use to store data (memory, bolt)")
+	flag.StringVar(&bucket, "bucket", "fakes3", "Bucket to create by default (required)")
 	flag.Parse()
 
-	faker := gofakes3.New(*db)
-	listenAndServe(*port, faker.Server())
+	if bucket == "" {
+		bucket = "fakes3"
```

Comment:
I really like to use the lighter approach first, thus this, but the history of this project was different and thus it could lead to unforeseen problems if people where expecting a persistent store. What's your opinion? Happy to take the lighter approach though.

Reply:
Do you mean defaulting to creating a bucket or defaulting to boltdb vs in-memory? I don't really have a preference either way personally. In the case of defaulting to bolt vs in-memory, one of my colleagues actually expressed surprise to me recently that our internal fork wasn't persistent; she said "Oh! I'm glad you told me, I was about to start putting huge files in there and expecting them to be on the disk!" The timing was quite lucky, I shudder to think what would've happened to her poor laptop once she started throwing a bunch of 2gb movie files into it!

Reply:
Ah, missed the line, yes I was referring to the boltdb vs in-memory ;-)
Hehe.. that would've been a good memory disaster